pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -86,6 +87,7 @@ def __init__(
         parallel_devices: Optional[List[torch.device]] = None,
         cluster_environment: Optional[ClusterEnvironment] = None,
         checkpoint_io: Optional[CheckpointIO] = None,
+        precision_plugin: Optional[PrecisionPlugin] = None,
```

Comment:
Why is it `Optional` here whereas it was required for the `Accelerator`?

This comment also applies to all other additions to each TTP.

Reply:
Optional for user who construct Strategy(), they can not pass percision_plugin which default is None. Then we set to default for them. If it's TPU or IPU we know which precision_plugin to set when it's None, if not we use precision_plugin().

If the precision plugin choiced by accelerator_connector, it guaranteed won't be None